File: app/src/main/java/io/maido/m8client/M8Application.java
package io.maido.m8client;

import android.app.Application;
import android.content.Context;
import android.os.StrictMode;

public class M8Application extends Application {

    private static Context mContext;

    @Override
    public void onCreate() {
        super.onCreate();
        mContext = this;
    }

    public static Context getContext() {
        return mContext;
    }

    public M8Application() {
        if (BuildConfig.DEBUG) {
            StrictMode.setVmPolicy(new StrictMode.VmPolicy.Builder()
                    .detectLeakedClosableObjects()
                    .penaltyLog()
                    .build());
        }
    }
}

File: app/src/main/kotlin/io/maido/m8client/M8Application.kt
package io.maido.m8client

import android.app.Application
import android.os.StrictMode
import android.os.StrictMode.VmPolicy

class M8Application : Application() {

    init {
        if (BuildConfig.DEBUG) {
            StrictMode.setVmPolicy(
                VmPolicy.Builder()
                    .detectLeakedClosableObjects()
                    .penaltyLog()
                    .build()
            )
        }
    }

}